src/main/java/com/libertymutualgroup/herman/aws/ecs/loadbalancing/EcsLoadBalancerV2Handler.java

```diff
@@ -213,26 +215,41 @@ public LoadBalancer createLoadBalancer(EcsClusterMetadata clusterMetadata, EcsPu
         String registeredUrl = urlPrefix + "." + definition.getService().getUrlSuffix();
         dnsRegistrar.registerDns(appName, "application", protocol, registeredUrl);
 
-        brokerDDoSWAFConfiguration(appName, protocol, elbScheme, loadBalancer);
+        brokerDDoSWAFConfiguration(appName, protocol, elbScheme, loadBalancer, definition);
 
         buildLogger.addLogEntry("ALB updates complete: " + loadBalancer.getLoadBalancerArn());
         return new LoadBalancer().withContainerName(containerName).withContainerPort(containerPort)
             .withTargetGroupArn(grp.getTargetGroupArn());
     }
 
     private void brokerDDoSWAFConfiguration(String appName, String protocol, String elbScheme,
-        com.amazonaws.services.elasticloadbalancingv2.model.LoadBalancer loadBalancer) {
+        com.amazonaws.services.elasticloadbalancingv2.model.LoadBalancer loadBalancer, EcsPushDefinition definition) {
         if (INTERNET_FACING.equals(elbScheme)
                 && HTTPS.equals(protocol)) {
             if (taskProperties.getDdosWaf() != null && taskProperties.getDdosWaf().getDdosWafLambda() != null) {
-                DdosWafBroker ddosWafBroker = new DdosWafBroker(buildLogger, taskProperties.getDdosWaf(), lambdaClient);
+                DdosWafBrokerProperties customProperties = getDdosWafBrokerPropertiesForApplication(definition);
+                DdosWafBroker ddosWafBroker = new DdosWafBroker(buildLogger, customProperties, lambdaClient);
```

Comment:
Does the WafBroker make sense to keep internal or separated out as an external broker? Feels like maybe an external broker to me, but you're better positioned to answer that @fritzdj.

Reply:
The WAF broker is separated out as an external broker now. However, this change was made so that we could define WAF conditions in the Herman template. As part of that we need to define the type (allow or block) for each condition: https://github.com/libertymutual/herman/blob/lfg/src/main/java/com/libertymutualgroup/herman/aws/ecs/broker/ddoswaf/WafRuleAction.java. Conditions tend to be app-specific.

Reply:
Cool, that works then 👍